src/plot_api/plot_api.js

```diff
@@ -326,16 +329,19 @@ Plotly.plot = function(gd, data, layout, config) {
             }
 
             // finally do all error bars at once
-            if(gd._fullLayout._hasCartesian) {
+            if(fullLayout._hasCartesian) {
                 ErrorBars.plot(gd, subplotInfo, cdError);
                 Lib.markTime('done ErrorBars');
             }
         }
 
-        // now draw stuff not on subplots (ie, pies)
-        // TODO: gotta be a better way to handle this
-        var cdPie = getCdModule(calcdata, Pie);
-        if(cdPie.length) Pie.plot(gd, cdPie);
+        // now draw stuff not on subplots (ie, only pies at the moment)
+        if(fullLayout._hasPie) {
+            var Pie = Plots.modules.pie._module;
```

Comment:
Should we maybe use `Plots.getModule('pie')` here instead for the sake of uniform access?